core/events.md

```diff
@@ -0,0 +1,520 @@
+# xRegistry Events
+
+## Abstract
+
+This specification defines the events that an xRegistry server MAY generate.
+
+## Table of Contents
+
+- [Overview](#overview)
+- [Notations and Terminology](#notations-and-terminology)
+  - [Notational Conventions](#notational-conventions)
+- [Event Definition](#event-definition)
+- [Entity Events](#entity-events)
+- [Sample xRegistry Interactions](#sample-xregistry-interactions)
+
+## Overview
+
+As updates are made to entities within an xRegistry instance, events SHOULD
+be generated that allows for interested parties to be notified of those
+changes. This specification defines the metadata associated with each event
+as [CloudEvent][https://cloudevents.io) metadata. Whether CloudEvents is used
+in the generation/serialization of the events is OPTIONAL, but it is
+RECOMENDED.
+
+This specification does not mandate the mechanisms by which events are sent
+to consumers, nor does it mandate how consumers registry interest in receiving
+events.
+
+Below is a sample event serialized as a "structured" CloudEvent due to a
+Group's `name` being changed:
+
+```yaml
+{
+  "specversion": "1.0",
+  "type": "io.xregistry.group.updated",
+  "source": "https://example.com/myRegistry",
+  "subject": "/dirs/d1",
+  "id": "A234-1234-1234",
+  "time": "2025-09-01T12:01:02Z",
+  "xregcorrelationid": "B9282-129301",
+  "data": {
+    "changed": [ "epoch", "modifiedat", "name" ]
+  }
+}
+```
+
+## Notations and Terminology
+
+### Notational Conventions
+
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD",
+"SHOULD NOT", "RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be
+interpreted as described in [RFC 2119](https://tools.ietf.org/html/rfc2119).
+
+In the pseudo JSON format snippets `?` means the preceding item is OPTIONAL,
+`*` means the preceding item MAY appear zero or more times, and `+` means the
+preceding item MUST appear at least once. The presence of the `#` character
+means the remaining portion of the line is a comment. Whitespace characters in
+the JSON snippets are used for readability and are not normative.
+
+Use of `<...>` the notation indicates a substitutable value where that is
+meant to be replaced with a runtime situational-specific value as defined by
+the word/phase in the angled brackets. For example `<NAME>` might be expected
+to be replaced by the "name" of the item being discussed.
+
+## Event Definition
+
+Notification of changes to the entities within an xRegistry instance MAY be
```

Comment:
Should this be a SHOULD?

Reply:
depends on how pushy we want to be :-)
I'll go with SHOULD for now.